sql_builder.go

```diff
@@ -27,6 +27,23 @@ func newSQLBuilder(q Query, m *Model, addColumns ...string) *sqlBuilder {
 	}
 }
 
+func hasLimitOrOffset(sqlString string) bool {
+	trimmedSQL := strings.TrimSpace(sqlString)
+	if matched, _ := regexp.MatchString("(?i).*\\s+limit\\s+[0-9]*(\\s?,\\s?[0-9]*)?$", trimmedSQL); matched {
+		return matched
+	}
+
+	if matched, _ := regexp.MatchString("(?i).*\\s+offset\\s+[0-9]*$", trimmedSQL); matched {
+		return matched
+	}
+
+	if matched, _ := regexp.MatchString("(?i).*\\s+rows only$", trimmedSQL); matched {
```

Comment:
Can you extract the regex outside the function and compile it only once? (using `MustCompile`)